integration-tests/conftest.py

```diff
@@ -1,10 +1,27 @@
 import pytest
+import subprocess
 
 
+@pytest.fixture(scope="session")
+def install_katello_rpm(test_config):
+    if "satellite" in test_config.environment:
+        # install katello rpm before register system against Satellite
+        cmd = "rpm -qa 'katello*' | xargs --no-run-if-empty rpm -e"
+        ret = subprocess.check_call(cmd, shell=True)
+        
+        satellite_hostname = test_config.get("candlepin", "host")
+        cmd = "rpm -Uvh http://%s/pub/katello-ca-consumer-latest.noarch.rpm" %satellite_hostname
+        ret = subprocess.check_call(cmd, shell=True)
+
 @pytest.fixture(scope="session")
 def register_subman(external_candlepin, subman_session, test_config):
-    subman_session.register(
-        username=test_config.get("candlepin", "username"),
-        password=test_config.get("candlepin", "password"),
-    )
+    activationkey=test_config.get("candlepin", "activation_keys")
+    org=test_config.get("candlepin", "org")
+    if activationkey and org:
+        subman_session.register(activationkey=activationkey, org=org)
```

Comment:
there are also environments supported in pytest-client-tools now, so you can add `environments=test_config.get("candlepin", "environments")` as parameter for this